packages/components/modules/messages/ChatRoomsList/ChatRoomItem/index.tsx

```diff
@@ -53,16 +57,43 @@ const ChatRoomItem: FC<ChatRoomItemProps> = ({
 
   const showBadge = room.unreadMessagesCount && room.unreadMessagesCount > 0
 
+  const [commit, isMutationInFlight] = useArchiveChatRoomMutation()
+
   return (
     <ActionsOverlay
       title="Chat"
       offsetTop={-12}
       actions={[
         {
-          disabled: true,
+          disabled: isMutationInFlight,
           icon: <ArchiveIcon />,
           label: 'Archive Chat',
-          onClick: () => {},
+          onClick: () => {
+            if (currentProfile?.id) {
+              commit({
+                variables: {
+                  input: {
+                    roomId: room.id,
+                    profileId: currentProfile.id,
+                    archive: !isInArchivedTab,
```

Comment:
Why do you need to pass this argument? Is this also unarchiving the chat (with the same archive icon)?

Reply:
I see, it is there because of the next PR.